src/stim/cmd/command_convert.cc

```diff
@@ -0,0 +1,567 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "stim/cmd/command_convert.h"
+
+#include <stdexcept>
+
+#include "command_help.h"
+#include "stim/arg_parse.h"
+#include "stim/dem/detector_error_model.h"
+#include "stim/io/measure_record_batch_writer.h"
+#include "stim/io/measure_record_reader.h"
+#include "stim/io/stim_data_formats.h"
+#include "stim/mem/simd_bits.h"
+
+using namespace stim;
+
+struct DataDetails {
+    int num_measurements;
+    int num_detectors;
+    int num_observables;
+    int bits_per_shot;
+    bool include_measurements;
+    bool include_detectors;
+    bool include_observables;
+};
+
+void process_num_flags(int argc, const char **argv, DataDetails *details_out) {
+    details_out->num_measurements = find_int64_argument("--num_measurements", 0, 0, INT64_MAX, argc, argv);
+    details_out->num_detectors = find_int64_argument("--num_detectors", 0, 0, INT64_MAX, argc, argv);
+    details_out->num_observables = find_int64_argument("--num_observables", 0, 0, INT64_MAX, argc, argv);
+
+    details_out->include_measurements = details_out->num_measurements > 0;
+    details_out->include_detectors = details_out->num_detectors > 0;
+    details_out->include_observables = details_out->num_observables > 0;
+}
+
+void process_dem(const char *dem_path, DataDetails *details_out) {
+    if (dem_path == nullptr) {
+        return;
+    }
+
+    FILE *dem_file = fopen(dem_path, "rb");
+    if (dem_file == nullptr) {
+        std::stringstream msg;
```

Comment:
You'd get this checking for free from `find_open_file_argument("--dem", nullptr, "rb", argc, argv)`

Reply:
I actually tried this originally, but `find_open_file_argument` throws an exception if this arg isn't passed (and this arg is optional): https://github.com/quantumlib/Stim/blob/60baacac26dfcc2a1e628f7426cc290d3d252266/src/stim/arg_parse.cc#L336-L340

We could extend `find_open_file_argument` to deal with optional FILE args if you think that'd be useful.

Reply:
Yeah that'd be pretty useful. Not needed for this PR, but in general the coupling of the default value to does-it-throw has been a thorn in those methods' utility.

Reply:
sure, I can follow up in a subsequent PR to add this and clean up any other uses where we have optional file args